Lib/typing.py

```diff
@@ -3296,3 +3297,23 @@ def method(self) -> None:
         # read-only property, TypeError if it's a builtin class.
         pass
     return method
+
+
+def get_protocol_members(tp: type, /) -> set[str]:
+    """Return the set of members defined in a Protocol.
+
+    Example::
+
+        >>> from typing import Protocol, get_protocol_members
+        >>> class P(Protocol):
+        ...     def a(self) -> str: ...
+        ...     b: int
+        >>> get_protocol_members(P)
+        {'a', 'b'}
+
+    Return None for arguments that are not Protocols.
+
+    """
+    if not getattr(tp, '_is_protocol', False) or tp is Protocol:
+        return None
```

Comment:
Not sure if we should return `None` or raise a `TypeError`? My instinct says we should raise, but I know `get_args` and `get_origin` both return `None` on invalid inputs.

Reply:
A TypeError would be unfriendly unless we also expose a way to check whether something is a Protocol first.

Reply:
I see what you mean. But it still feels "wrong" not to raise on invalid input here.

I wouldn't mind a `typing.is_protocol` function actually, similar to `typing.is_typeddict` (which is already a thing).